docs/cookbook/adding-a-new-elasticsearch-index.md

```diff
@@ -0,0 +1,294 @@
+
+```diff
++   /**
++    * @var \Shopsys\FrameworkBundle\Model\Category\CategoryRepository
++    */
++   protected $categoryRepository;
++
++   /**
++    * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
++    */
++   protected $domain;
++
++   /**
++    * @param \Shopsys\FrameworkBundle\Model\Category\CategoryRepository $categoryRepository
++    * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
++    */
++   public function __construct(CategoryRepository $categoryRepository, Domain $domain)
++   {
++       $this->categoryRepository = $categoryRepository;
++       $this->domain = $domain;
++   }
++
+```
+
+When we have injected services we may implement `getTotalCount()`
+
+```diff
+   public function getTotalCount(int $domainId): int
+    {
+-       // TODO: Implement getTotalCount() method.
++       return count($this->categoryRepository->getTranslatedVisibleSubcategoriesByDomain(
++           $this->categoryRepository->getRootCategory(),
++           $this->domain->getDomainConfigById($domainId)
++       ));
+    }
+```
+
+and also a `getExportDataForBatch()` with a private converting method `convertToElastic()`
+
+```diff
+    public function getExportDataForBatch(int $domainId,int $lastProcessedId,int $batchSize) : array
+    {
+-       // TODO: Implement getExportDataForBatch() method.
++       $domainConfig = $this->domain->getDomainConfigById($domainId);
++       $categories = $this->categoryRepository->getTranslatedVisibleSubcategoriesByDomain(
++           $this->categoryRepository->getRootCategory(),
++           $domainConfig
++       );
++       $locale = $domainConfig->getLocale();
++       foreach ($categories as $category) {
++           $result[$category->getId()] = $this->convertToElastic($category, $domainId, $locale);
++       }
++       return $result;
+    }
++
++   private function convertToElastic(Category $category, int $domainId, string $locale): array
++   {
++       return [
++           'name' => $category->getName($locale),
++           'description' => $category->getDescription($domainId),
++           'parentId' => $category->getParent()->getId(),
++           'level' => $category->getLevel(),
++           'uuid' => $category->getUuid(),
++       ];
++   }
+```
+
+!!! note
+The `getExportDataForBatch()` must return serialized array of rows indexed by its ID
+
+Now we can export categories data (name, description, parentId, level, and uuid) into elasticsearch index via `./phing elasticsearch-export -D elaticsearch.index=category` (we need to have created index first, see the step above).
+
+### Exporting via cron
+
+We may automate the export process via CronModule which is super easy.
+All we need to achieve this goal is to create a new class `CategoryExportCronModule` which extends `AbstractExportCronModule`.
```

Comment:
it would be nice to say the best place where to create the new `CategoryExportCronModule` (it's later guessable from namespace, but at first it made me thinking)